unstructured/ingest/connector/outlook.py

```diff
@@ -0,0 +1,214 @@
+import hashlib
+import os
+from collections import defaultdict
+from dataclasses import dataclass, field
+from itertools import chain
+from pathlib import Path
+from typing import List
+
+from office365.onedrive.driveitems.driveItem import DriveItem
+
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+MAX_NUM_EMAILS = 10000
```

Comment:
what drives this max number? does the library trigger paging otherwise? is there a generator or some alternative so we can avoid the upper limit?

Reply:
The library triggers paging as the default. Which is 10. Supposedly there is a timeout at some point. But consider that this call is just getting mail objects (It's not downloading), it should be pretty fast.

At the point that it starts downloading, it does it with separate calls, so there shouldn't be a timeout. Though it would be hard for us to test with just a couple emails...

Reply:
got it, makes sense. is there any reason we need to limit this at all?

Reply:
If we don't use `.top()` (line 209) it will default to 10. if we use `.top()` with nothing in it. I think it always defaults to 10 also. I will look in the documentation.

Reply:
Looks like top needs an parameter number. At least there is no default in the documentation. Without .top you only get 10 documents

Reply:
that's so bizarre! issue is that's the pre-filtered number, so that will include you're entire inbox (I'm guessing trash too?). I guess I'm tempted to make this something much bigger, like 10M. kind of a bummer we have to query all of the full messages and hold them in memory, unlike other connectors where each doc is responsible for actually fetching content, bc that does have me slightly worried about things blowing up and waiting for a giant call before processing. 

fwiw it [looks like](https://github.com/vgrem/Office365-REST-Python-Client/blob/2b603f5c9643ece12b40cfcaf1e4005cbca141e1/examples/outlook/messages/list_new.py#L12) there's a way to filter in that original query by mailbox, right? if that's correct, we should probably do that for the sake of reducing memory and request demands and skirting around the max fetch a bit.

Reply:
ooh. let me look into that.

Reply:
I'll look deeper into this. For some reason I am recalling that a few of the root folders they allow "Inbox" as the ID (and "Sent Items" and a few others). There may have been a reason that this did not work though in the long run for non standard folders. I can't totally remember. But I will definitely do some exploration.

Reply:
Ok. It will only take folder IDs in that mail_folder. BUT we could get all the folders and subfolders first (and their ids) and then collect the mail after that we get the folders. Then we don't have to do any filtering later. And we will only get the relevant emails.

Glad you thought of doing that.

Should be a reasonably easy fix.

Reply:
Awesome! Thanks for digging here.